### PR TITLE
backup: abstract file backend as backend interface

### PR DESCRIPTION
**Pull request by xiang90**
prepare for a s3 backend.

### CODE REVIEW

File: pkg/backup/http.go [hunk 1]
```diff
@@ -87,5 +87,9 @@ func (b *Backup) serveSnap(w http.ResponseWriter, r *http.Request) {
 	if r.Method == http.MethodHead {
 		return
 	}
-	http.ServeFile(w, r, path.Join(b.backupDir, fname))
+
+	_, err = io.Copy(w, rc)
+	if err != nil {
+		logrus.Errorf("failed to write backup to %s: %v", r.RemoteAddr, err)
```
Comment (hongchaodeng) [inline pkg/backup/http.go]: What happen to http response on this case?

Comment (xiang90) [inline pkg/backup/http.go]: io copy fails. http connection probably fails. we do nothing i guess. but we can try to write a error to http.

Comment (xiang90) [inline pkg/backup/http.go]: actually we cannot write http header as long as we ever write one byte in the body.

Comment (xiang90) [inline pkg/backup/http.go]: so it does not worth it.

**Comment by xiang90** [comment]
@hongchaodeng Any more comments or concerns that I should address?

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**